hypervisor/common/hypercall.c

```diff
@@ -166,21 +166,35 @@ int32_t hcall_create_vm(struct acrn_vm *vm, uint64_t param)
 			/* Filter out the bits should not set by DM and then assign it to guest_flags */
 			vm_config->guest_flags |= (cv.vm_flag & DM_OWNED_GUEST_FLAG_MASK);
 
-			/* GUEST_FLAG_RT must be set if we have GUEST_FLAG_LAPIC_PASSTHROUGH set in guest_flags */
-			if (((vm_config->guest_flags & GUEST_FLAG_LAPIC_PASSTHROUGH) != 0U)
-				&& ((vm_config->guest_flags & GUEST_FLAG_RT) == 0U)) {
-				pr_err("Wrong guest flags 0x%lx\n", vm_config->guest_flags);
+			/* post-launched VM is allowed to assign PCPU from vm_config->pcpu_affinity_bitmap only */
+			if ((cv.pcpu_bitmap & ~(vm_config->pcpu_affinity_bitmap)) != 0UL) {
+				pr_err("%s: Post-launch VM can't share PCPU with Pre-launch VM!", __func__);
 			} else {
-				if (create_vm(vm_id, vm_config, &target_vm) != 0) {
-					dev_dbg(DBG_LEVEL_HYCALL, "HCALL: Create VM failed");
-					cv.vmid = ACRN_INVALID_VMID;
-				} else {
-					/* return a relative vm_id from SOS view */
-					cv.vmid = vmid_2_rel_vmid(vm->vm_id, vm_id);
-					cv.vcpu_num = vm_config->vcpu_num;
+				/* DM could overwrite the statically configured PCPU bitmap */
+				if (bitmap_weight(cv.pcpu_bitmap) != 0U) {
+					vm_config->vcpu_num = bitmap_weight(cv.pcpu_bitmap);
+					vm_config->pcpu_affinity_bitmap = cv.pcpu_bitmap;
 				}
 
-				ret = copy_to_gpa(vm, &cv, param, sizeof(cv));
+				/*
+				 * GUEST_FLAG_RT must be set if we have GUEST_FLAG_LAPIC_PASSTHROUGH
+				 * set in guest_flags
+				 */
+				if (((vm_config->guest_flags & GUEST_FLAG_LAPIC_PASSTHROUGH) != 0U)
```

Comment:
use is_lapic_pt_configured().

Reply:
Check previous patch too for this issue.

Reply:
At this moment, the vm hasn't been created, while is_lapic_pt_configured(vm) makes use of get_vm_config(vm->vm_id), which is not available yet. BTW, guest_flags is of uint64_t, will change 0U  to 0UL.

Reply:
ok